lib/airbrake/passenger/integration.rb

```diff
@@ -0,0 +1,59 @@
+module Airbrake
+  # Integration for Phusion Passenger
+  module Passenger
+    IntegrationError = Class.new(::StandardError)
+    class << self
+      def notifier
+        @notifier || :default
+      end
+
+      def notifier=(val)
+        if val.is_a?(Symbol)
+          @notifier = val
+        else
+          @notifier = val.to_sym
+        end
+      end
+    end
+    # Implementation of Passenger Integration for Versions 4.0 thru 5.0
+    module Integration
+      VERSION_REQUIREMENT = Gem::Requirement.new('>= 4.0', '< 5.1').freeze
+      class << self
+        def integrate!
+          return @integrated if defined?(@integrated)
+          raise IntegrationError, "Your version of PhusionPassenger " \
+            "is not currently supported by the airbrake gem." unless compatible?
+          unless defined?(::PhusionPassenger::LoaderSharedHelpers)
+            ::PhusionPassenger.require_passenger_lib 'loader_shared_helpers'
+          end
+          ::PhusionPassenger::LoaderSharedHelpers.send :extend, ClassMethods
+        rescue IntegrationError => e
+          Kernel.warn e
+          ::Airbrake.notify_sync(e, {}, notifier)
```

Comment:
@kyrylo Hmm.. is it OK to check and see if they `require`d `airbrake` manually here (as opposed to `bundler` doing it) and if so raise the exception instead of just warning it?

Because if someone has required `airbrake` prior to the `bundler` invocation, it's all but guaranteed they're trying to set up `airbrake` integration so that it can catch all exceptions. Throwing a hard error will at least make it obvious something went wrong, and `passenger` (in dev mode at least) will display a friendly error page with a backtrace. Otherwise it's at least a huge hint to check the trace in the logs.